thirtyinch-rx2/src/main/java/net/grandcentrix/thirtyinch/rx2/DisposableHandlingPresenter.kt

```diff
@@ -0,0 +1,16 @@
+package net.grandcentrix.thirtyinch.rx2
+
+import net.grandcentrix.thirtyinch.TiPresenter
+import net.grandcentrix.thirtyinch.TiView
+
+/**
+ * Convenience base [TiPresenter] class implementing [DisposableHandler] to be extended by your presenters.
+ *
+ * Makes possible to use [Disposable][io.reactivex.disposables.Disposable] handling extension functions without the
+ * need to implement the interface yourself.
+ */
+open class DisposableHandlingPresenter<V : TiView> : TiPresenter<V>(), DisposableHandler {
```

Comment:
❌
I'm totally against this!
Please think from the "new user perspective":
* Why are there two Presenters?
* Do I have to use the DIsposableHandlingPresenter as soon as I use the rx2 dependency?

Something like this leads just to confusing instead of helping the user...

Reply:
Yes, valid point. I was also not a fan of this, as it breaks the whole principle of composition over inheritance.